src/kokkos/test/kokkos/testEigenGPU.cc

```diff
@@ -0,0 +1,403 @@
+                                                     Kokkos::View<float*, KokkosExecSpace> vhits_ge,
+                                                     const int& i) {
+  double* __restrict__ phits = vhits.data();
+  float* __restrict__ phits_ge = vhits_ge.data();
+
+  Rfit::Map3xNd<N> hits(phits + i, 3, N);
+  Rfit::Map6xNf<N> hits_ge(phits_ge + i, 6, N);
+  hits_ge = MatrixXf::Zero(6, N);
+  fillHitsAndHitsCov(hits, hits_ge);
+}
+
+template <int N>
+void testFit() {
+  constexpr double B = 0.0113921;
+
+  Kokkos::View<double*, KokkosExecSpace> d_hits("d_hits", Rfit::maxNumberOfTracks() * sizeof(Rfit::Matrix3xNd<N>));
+  Kokkos::View<float*, KokkosExecSpace> d_hits_ge("d_hits_ge", Rfit::maxNumberOfTracks() * sizeof(Rfit::Matrix6xNf<N>));
+  Kokkos::View<double*, KokkosExecSpace> d_fast_fit_results("d_fast_fit_results",
+                                                            Rfit::maxNumberOfTracks() * sizeof(Vector4d));
+  Kokkos::View<Rfit::line_fit*, KokkosExecSpace> d_line_fit_results("d_line_fit_results",
+                                                                    Rfit::maxNumberOfTracks() * sizeof(Rfit::line_fit));
+  Kokkos::View<Rfit::circle_fit*, KokkosExecSpace> d_circle_fit_results(
+      "d_circle_fit_results", Rfit::maxNumberOfTracks() * sizeof(Rfit::circle_fit));
+
+  Rfit::Matrix3xNd<N> hits;
+  Rfit::Matrix6xNf<N> hits_ge = MatrixXf::Zero(6, N);
+
+  double* fast_fit_resultsGPUret = new double[Rfit::maxNumberOfTracks() * sizeof(Vector4d)];
+  Rfit::circle_fit* circle_fit_resultsGPUret = new Rfit::circle_fit();
+  Rfit::line_fit* line_fit_resultsGPUret = new Rfit::line_fit();
+
+  fillHitsAndHitsCov(hits, hits_ge);
+
+  std::cout << "sizes " << N << ' ' << sizeof(hits) << ' ' << sizeof(hits_ge) << ' ' << sizeof(Vector4d) << ' '
+            << sizeof(Rfit::line_fit) << ' ' << sizeof(Rfit::circle_fit) << std::endl;
+
+  std::cout << "Generated hits:\n" << hits << std::endl;
+  std::cout << "Generated cov:\n" << hits_ge << std::endl;
+
+  // FAST_FIT_CPU
+  Vector4d fast_fit_results;
+#ifdef USE_BL
+  BrokenLine::BL_Fast_fit(hits, fast_fit_results);
+#else
+  Rfit::Fast_fit(hits, fast_fit_results);
+#endif
+  std::cout << "Fitted values (FastFit, [X0, Y0, R, tan(theta)]):\n" << fast_fit_results << std::endl;
+
+  // cudaMemset d_fast_fit_results & d_line_fit_results to 0
+  Kokkos::deep_copy(KokkosExecSpace(), d_fast_fit_results, 0);
+  // Kokkos::deep_copy(KokkosExecSpace(), d_line_fit_results, 0) will result in compilation error:
+  // no instance of overloaded function "Kokkos::deep_copy" matches the argument list. argument
+  // types are: (KokkosExecSpace, Kokkos::View<kokkos_cuda::Rfit::line_fit *, KokkosExecSpace>, int).
+  // Use for loop instead
+  auto tmpN = Rfit::maxNumberOfTracks() * sizeof(Rfit::line_fit);
+  Kokkos::parallel_for(
+      "init_line_fit_res",
+      Kokkos::RangePolicy<KokkosExecSpace>(KokkosExecSpace(), 0, tmpN),
```

Comment:
Without the `reinterpret_cast` this should be changed to something along
```suggestion
  Kokkos::parallel_for(
      "init_line_fit_res",
      Kokkos::RangePolicy<KokkosExecSpace>(KokkosExecSpace(), 0, Rfit::maxNumberOfTracks()),
```
?

Reply:
Oops, corrected. The good thing is I ran the code and it didn't crash.